src/modules/infobox/components/ImportContent.vue

```diff
@@ -0,0 +1,254 @@
+                // Switch focus to the first item in the list that can get it.
+                // Initially, this will be the first entry of the location list.
+                this.$refs.providers.querySelector('[tabindex="0"]').focus()
+            })
+        },
+        goToPrevious(currentKey) {
+            if (currentKey === 0) return
+            const key = currentKey - 1
+            this.$refs.providers.querySelector(`[tabindex="${key}"]`).focus()
+        },
+        goToNext(currentKey) {
+            if (currentKey >= this.filteredList.length - 1) return
+            const key = currentKey + 1
+            this.$refs.providers.querySelector(`[tabindex="${key}"]`).focus()
+        },
+        goToFirst() {
+            this.$refs.providers.querySelector('[tabindex="0"]').focus()
+        },
+        goToLast() {
+            this.$refs.providers
+                .querySelector(`[tabindex="${this.filteredList.length - 1}"]`)
+                .focus()
+        },
+        handleFileContent(fileContent, url) {
+            if (isWmsGetCap(fileContent)) {
+                const parser = new WMSCapabilities()
+                const getCap = parser.read(fileContent)
+                // todo just for test
+                this.importedLayers = getCap.Capability.Layer.Layer.map((l) =>
+                    getCapWMSLayers(getCap, l)
+                ).filter((l) => !!l)
+            } else if (isWmtsGetCap(fileContent)) {
+                const parser = new WMTSCapabilities()
+                const getCap = parser.read(fileContent)
+                // todo just for test
+                this.importedLayers = getCap.Contents.Layer.map((l) =>
+                    getCapWMTSLayers(url, getCap, l)
+                ).filter((l) => !!l)
+            } else if (isKml(fileContent)) {
+                // todo just for test
+                this.importedLayers = [new KMLLayer(url, true, 1, null, null, null, null, true)]
+            } else if (isGpx(fileContent)) {
+                // TODO GPX layer not done yet
+            } else {
+                throw new Error('Wrong file content.')
+            }
+        },
+        async handleFileUrl() {
+            const url = await transformUrl(this.importValue, this.lang)
+            try {
+                const response = await fetch(url)
+                const fileContent = await response.text()
+                this.handleFileContent(fileContent, url)
+                this.uploadBtnStatus = 'succeeded'
+                setTimeout(() => (this.uploadBtnStatus = 'default'), BTN_RESET_TIMEOUT)
+            } catch (e) {
+                console.error(e)
+                this.uploadBtnStatus = 'failed'
+                setTimeout(() => (this.uploadBtnStatus = 'default'), BTN_RESET_TIMEOUT)
+            }
+        },
+    },
+}
+</script>
+
+<style lang="scss" scoped>
+.import-overlay-content {
+    height: 250px;
+}
+.connect-btn {
+    width: 100%;
+    margin-top: 5px;
```

Comment:
better use the bootstrap padding and margin classes

Reply:
updated